### DISCUSSION TITLE
cQube Schema requirements for Assessments

### DISCUSSION THREAD

**Discussion by ananya-jain1** [Sunbird-Saral/Community#67] [General]
Here is the schema in which cQube accepts data for assessments - [Link](https://project-sunbird.atlassian.net/wiki/spaces/CQUB/pages/3279290379/cQube+v5.0+Beta+-+Schema+Documentation#Student-Assessment-Program). Please navigate to the 'input schema' heading under 'Student Assessments' program.

Would be great if SARAL can also emit data in a similar format so that the linkage for states between SARAL & cQube becomes seamless. 

We can leverage this discussion thread in case of any questions with respect to the schema.

@dileep-gadiraju

**Comment by dileep-gadiraju** [comment]
Hi @ananya-jain1 ,

Fields I found(like total_marks, total_obtained_marks) in the below link are extracted after the correction process in the CQube. These data elements can't be extracted from Saral side. please check and confirm once.
https://project-sunbird.atlassian.net/wiki/spaces/CQUB/pages/3279290379/cQube+v5.0+Beta+-+Schema+Documentation#Input-Schema.5

**Comment by ananya-jain1** [comment]
Hi @dileep-gadiraju - I didn't understand what you meant by 'correction process in cQube'. Could you please throw some light on this?

**Reply by dileep-gadiraju** [reply]
Hi @ananya-jain1 , i spoke to shryashi today about the same. fields **total_marks, total_obtained_marks** are populated outside Sunbird saral processing. Saral captures the options the student or teacher chooses on the physical answer sheet but the correction of these answer sheets or scoring for these questions is done outside the scope of Saral. For example, a student with student Id 100202232 chooses A,B,A,C,A,A,B........ etc for 1 to 30 questions, So saral captures this data that is filled by the student and the exam correction by assigning marks for each of these questions against the answer key is done outside Saral. So Saral can emit data with fields studentid , exam, subject, date, Option Chosen for Q1, Option Chosen for Q2 etc.... but not the actual marks secured by each student.

**Reply by ananya-jain1** [reply]
Got it Dileep, I also had a conversation with Shreyashi on the same. 

We can emit the data in the specific schema for 'LO-related information' (2nd table under [input schema](https://project-sunbird.atlassian.net/wiki/spaces/CQUB/pages/3279290379/cQube+v5.0+Beta+-+Schema+Documentation#Input-Schema.5)) right? I believe that data is captured in SARAL.

**Comment by dileep-gadiraju** [comment]
Hi @ananya-jain1 ,

Exam metadata/master data we set up in SARAL backend has the below fields as of today. examLO, indicatorTitle attribute can carry LO info. If this is set up appropriately, we can look up this attribute for a given exam.

```json
{
    "subject": "Hindi",
    "examLO": "Hindi",
    "classId": "2",
    "examDate": "23/09/2021",
    "totalMarks": 20,
    "type": "hindi_4s_20q_omr",
    "questions": [
      {
        "questionId": "ROLLNUMBER1_Q1",
        "tags": [
          {
            "tagName": "historic",
            "selected": false
          }
        ],
        "indicatorTitle": "L0-1",
        "questionMarks": "1"
      }
```
